src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -603,17 +605,28 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                     CurlHandler.EventSourceTrace("Size: {0}", size, easy: easy);
                     try
                     {
-                        // The callback is invoked once per header; multi-line headers get merged into a single line.
-                        string responseHeader = Marshal.PtrToStringAnsi(buffer).Trim();
                         HttpResponseMessage response = easy._responseMessage;
 
-                        if (!TryParseStatusLine(response, responseHeader, easy))
+                        CurlResponseHeaderReader reader = new CurlResponseHeaderReader(buffer, size);
```

Comment:
Let me do some more testing to see what curl hands back for non-ASCII header values. Probably won't get to it until later today.